beacon_node/beacon_chain/src/attestation_rewards.rs

```diff
@@ -0,0 +1,183 @@
+use crate::{BeaconChain, BeaconChainError, BeaconChainTypes};
+use eth2::lighthouse::attestation_rewards::{IdealAttestationRewards, TotalAttestationRewards};
+use eth2::lighthouse::StandardAttestationRewards;
+use participation_cache::ParticipationCache;
+use safe_arith::SafeArith;
+use slog::{debug, Logger};
+use state_processing::{
+    common::altair::BaseRewardPerIncrement,
+    per_epoch_processing::altair::{participation_cache, rewards_and_penalties::get_flag_weight},
+};
+use std::collections::HashMap;
+use store::consts::altair::PARTICIPATION_FLAG_WEIGHTS;
+use types::consts::altair::WEIGHT_DENOMINATOR;
+
+use types::{Epoch, EthSpec};
+
+impl<T: BeaconChainTypes> BeaconChain<T> {
+    pub fn compute_attestation_rewards(
+        &self,
+        epoch: Epoch,
+        mut validators: Vec<usize>,
+        log: Logger,
+    ) -> Result<StandardAttestationRewards, BeaconChainError> {
+        debug!(log, "computing attestation rewards"; "epoch" => epoch, "validator_count" => validators.len());
+
+        // Get state
+        let spec = &self.spec;
+
+        let state_slot = (epoch + 1).end_slot(T::EthSpec::slots_per_epoch());
+
+        let state_root = self
+            .state_root_at_slot(state_slot)?
+            .ok_or(BeaconChainError::UnableToFindTargetRoot(state_slot))?;
+
+        let state = self
+            .get_state(&state_root, Some(state_slot))
+            .and_then(|maybe_state| {
+                maybe_state.ok_or(BeaconChainError::MissingBeaconState(state_root))
+            })?;
```

Comment:
We can replace this `and_then` by `?`. The `?` is really a shorthand for `and_then`, so writing `and_then` explicitly is usually not necessary.

```suggestion
            .get_state(&state_root, Some(state_slot))?
            .ok_or(BeaconChainError::MissingBeaconState(state_root))?;
```